fulltext/__init__.py

```diff
@@ -326,13 +327,30 @@ def backend_from_fname(name):
     ext = splitext(name)[1]
     try:
         mime = EXTS_TO_MIMETYPES[ext]
+
     except KeyError:
-        with open(name, 'rb') as f:
-            return backend_from_fobj(f)
+        try:
+            f = open(name, 'rb')
+
+        except IOError as e:
+            # The file may not exist, we are being asked to determine it's type
+            # from it's name. Other errors are unexpected.
+            if e.errno != errno.ENOENT:
+                raise
```

Comment:
> backend_from_fname() should not assume the file exists.

I think this should be the case only in case `name` arg is provided, not if `path` (the first arg) is provided. With this change `get("file-that-does-not-exist")` will succeed and return an empty string, am I correct?

Reply:
https://github.com/Veristack/duster/issues/49
why don't we just make `duster` catch `IOError / ENOENT` instead?

Reply:
```python
>>> import fulltext
>>> fulltext.get('file-that-does-not-exist')
fulltext/util.py:126: UserWarning: don't know how to handle 'file-that-does-not-exist'; assume 'application/octet-stream'
  warnings.warn(msg, UserWarning)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "fulltext/__init__.py", line 520, in get
    encoding_errors=encoding_errors, _wtitle=_wtitle)
  File "fulltext/__init__.py", line 481, in _get
    text = fun(inst, path_or_file)
  File "fulltext/__init__.py", line 265, in handle_path
    with open(path, 'rb') as f:
IOError: [Errno 2] No such file or directory: 'file-that-does-not-exist'
>>> 
```